lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -2135,7 +2139,17 @@ def cmd_unset(*args)
   #   stage since the command itself has been completed.
   def cmd_unset_tabs(str, words)
     datastore = active_module ? active_module.datastore : self.framework.datastore
-    datastore.keys
+    keys = datastore.keys
+
+    mod = active_module
+    if mod
+      mod.options.each do |name, mod_opt|
+        next if Msf::OptCondition.show_option(mod, mod_opt)
+        i = keys.index(name)
+        keys.delete_at i if i
+      end
```

Comment:
This can be made a little clearer with `Array#delete_if`
```suggestion
      keys = keys.delete_if do |name|
        !Msf::OptCondition.show_option(mod, mod.options[name])
      end
```

Reply:
Changed in e1b4d17de6c347cfa70cbb5946a0a5bf8b72f3ff. I had to tweak it just slightly to `!(mod_opt = mod.options[name]).nil? && !Msf::OptCondition.show_option(mod, mod_opt)` to avoid an error where the option was nil.